include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -84,8 +84,29 @@ A_{3, 2}  A_{3, 3}  A_{3, 0}  A_{3, 1} ...   [phase 1] /
         auto mfmaEnc = dotOpEnc.getParent().dyn_cast<MfmaEncodingAttr>();
 
         if (mfmaEnc) {
-          // Swizzling is currently disabled for MFMA
-          return $_get(context, 1, 1, 1, order);
+          int kDimNum = dotOpEnc.getOpIdx() == 0 ? 1 : 0;
+          bool isKDimInner = (order[0] == kDimNum);
+          if (isKDimInner) {
+            const int numBanks = 32;
+            const int bankBitWidth = 32;
+
+            // number of inner dimension rows per one pattern repeat
+            int outerDimGranularity = mfmaEnc.getNonKDim();
+            int typeBitWidth = eltTy.getIntOrFloatBitWidth();
+            int innerDimLength = shape[order[0]];
+            int elemsPerOneBanksRow = (numBanks * bankBitWidth) / typeBitWidth;
+
+            int perPhase = std::max(1, elemsPerOneBanksRow / innerDimLength);
+            int maxPhase = outerDimGranularity / perPhase;
+            int vecSize = innerDimLength / maxPhase;
```

Comment:
`vecSize` is used as the vector store length during `convert_layout #blocked -> #shared`. Therefore, it should be limited by `128/typeBitWidth`.

Actually, I think we can be aggressive to set `vecSize` = 128/typeBitWidth so that `ds_write_b128` is always used.

Reply:
Interesting idea, but I am afraid, we can not be that aggressive.

Consider case fp16, matrix 32x64 (mxk):
vecSize = 128 / 16 = 8
perPhase = 1
maxPhase - could be at most 8 (vecSize * maxPhase <= k), which is not enough to eliminate all bank conflicts

Reply:
@zhanglx13  How do you figure maxphase of 16 in this case?

Reply:
@jayfurmanek 
It should be 16 based on the following facts:
- 1 row of the matrix (32x64) takes up 32 banks (64 x f16 element / 4 bytes per bank), this means 1 phase (32 banks) contains 1 row of the matrix
- At every cycle, 16 workitems need to load 16 x 4 elements (or 8), which spread across 16 rows, of the matrix in LDS.
- We want the elements loaded from different rows to be at different banks to avoid bank conflicts. 

Therefore, we need the 16 rows to swizzle in different patterns. Since rows in the same phase are swizzled in the same pattern (in this example, one phase contains only one row), we need at least 16 different phases.

Reply:
@binarman I think you are right. We are not able to avoid bank conflicts when `vecSize` = 8.
More specifically, for f16 data type, we must have
- `BLOCK_K/vecSize >= maxPhase` so that colSwizzled won't be out of bound
- `maxPhase*64/BLOCK_K >= 16` so that the number of phases can cover the 16 rows

This leads to `BLOCK_K/vecSize >= maxPhase >= BLOCK_K/4`, which is solvable when `vecSize <= 4`. So in this PR, I think we can set `vecSize` = 4 and `maxPhase` = BLOCK_K/4 to avoid bank conflits.

Reply:
> We can have a discussion later this week

Great!
I've experimented a little with different vec sizes, and got some contradictory results:

if `maxPhase` set to 16, vec size becomes larger.
I got **less** bank conflicts per wave in omniperf statistics, but overall execution time grows. I think this is a side effect of write bank conflicts you mentioned during some layout conversions.

I'll try to investigate it a bit closer soon.

Reply:
I experimented a bit as well and was seeing similar things. It didn't really move the numbers. I'd like to merge this as is for now. We can adjust in further PRs as needed.

Reply:
Good discussion!